### PR TITLE
YALB-79: Text-with-image component

### PR DESCRIPTION
**Pull request by ModulesUnraveled**
### Description of work
- Creates the Text with Image component
- Adds it as an option for the "Intro Content" on the example Standard Page

### Testing Link(s)
- [ ] Navigate to the [Text with Image Story](https://deploy-preview-XX--dev-component-library-twig.netlify.app/?path=/story/molecules-text-with-image--text-with-image)
- [ ] Navigate to the [Text with Image Story](https://deploy-preview-XX--dev-component-library-twig.netlify.app/?path=/story/page-examples-page-examples--standard-page)

### Functional Review Steps
- [ ] Verify the component controls all work as expected
- [ ] Verify you can add it to the Standard page Intro section (I only added options for the two widths so it wouldn't make the list overwhelming)

### Design Review
- [ ] Verify the designs match the [Figma Designs](https://www.figma.com/file/OVfmCW5s1gI7xnrM8ibX2y/UI-Kit-Alpha-%5BYaleSites%5D)

### Accessibility Review
- [ ] Verify the component meets Accessibility requirements. I'm particularly curious about the image, since it can switch positions (via css). I don't think it's focusable, so that shouldn't matter... I wasn't sure if there was something else to consider.

### CODE REVIEW

File: components/02-molecules/text-with-image/text-with-image.twig [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      {% include "@atoms/typography/headings/heading.twig" with {
+        heading: text_with_image__heading,
+        heading__level: '2',
+        heading__blockname: text_with_image__base_class,
+      } %}
+      {# Subheading #}
+      {% if text_with_image__subheading %}
+        {% include "@atoms/typography/headings/heading.twig" with {
+          heading: text_with_image__subheading,
+          heading__level: '3',
```
Comment (kara-franco) [inline components/02-molecules/text-with-image/text-with-image.twig:47]: Looking really good!! 

The subheading doesn't need to be semantically a heading, it's not a new section or subsection of the parent heading. 

For a screen reader user, having the dates as H3's will clutter their web rotor (a tool that they use pull out headings/landmarks/links in the DOM, shown below). 

Screen Shot 2022-03-11 at 9 58 19 AM

**Review by kara-franco — APPROVED** [review]

**Review by kara-franco — CHANGES_REQUESTED** [review]
Just need to update the subheading and all is looking good!

**Comment by ModulesUnraveled** [comment]
@kara-franco I removed the "h3" tag so it won't show up as an "official" heading

**Review by kara-franco — APPROVED** [review]
Yay, looks (and sounds) great!

**Comment by ahitson** [comment]
Will there be image options for this?

**Review by callinmullaney — APPROVED** [review]
@ModulesUnraveled This looks great to me. I didn't see anything that stood out to me that should be addressed. I really like the control options/variations for this component :tada:

**Comment by ModulesUnraveled** [comment]
@ahitson I don't think there's anything for us to do on the frontend as far as aspect ratio goes. On the backend, we'll need to make sure we address it as a requirement. That's where the user would choose the crop ratio. But the frontend component should accommodate whatever choice they make.

**Review by ahitson — APPROVED** [review]

**Pull request merged**